### ISSUE TITLE
Play 2 videos synchronously locally on 1 computer (Kiosk)

### ISSUE THREAD

**Issue by Remonell** [caprica/vlcj#723]
Hi,
I was trying to start 2 videos at the same time with 2 separate instances of a MediaPlayerComponent in 2 separate Frames.
1 video is in FullHD and the other in UHD. The problem is, that the FullHD video starts instantly while the UHD video takes a little bit to start. And in the waiting time at the start of the UHD video it just shows a black screen which I need to get rid of. Instead it should show the last frame of the prior video (playing diffrent videos on button press) until the new UHD video is properly loaded and can play. 

I was trying to wait until mediaPlayer.isPlaying() is true of the UHD instance. But it seems this variable becomes true in the buffering phase so this didnt solve my problem.

```
boolean playing= uhdMediaPlayer.isPlaying();
while(!playing) {
    playing= uhdMediaPlayer.isPlaying();
    System.out.println(count+" UHD is playing="+playing);
    count++;
}
mediaPlayer.playMedia(fhdFilePath);
```

I dont know if this is the right place to ask for suggestions on this matter. 

Kind regards

**Comment by caprica** [comment]
Did you try and wait until buffering reports 100%?

You could also try and wait for the "media player ready event" if you're using the most recent release of vlcj, and if not try waiting for the first "position changed" event.

Also don't use tight loops like that, use events.

**Comment by Remonell** [comment]
Hey caprica, 
thanks for the quick response. 
I did not check for the 100% buffering yet. I will try it out though. I will also try the event listener approach.

**Comment by Remonell** [comment]
The mediaPlayerReady Event was the right hint. It triggers in the same moment when my UHD player shows the first frame of the video. Thanks again!

Now I just need to get rid of the black phase while the uhd mediaPlayer is (probably) buffering the large video. Do you have any suggestion or hint for that problem? Ideally the player shows the last frame of the previous video until the new video is buffered completely and then shows the first frame of the new video. Like that there would not be black frames inbetween the videos.

I have been trying around the last few hours with following approach. On button click I create a new EmbeddedMediaPlayerComponent(EMPC) and also a new MediaPlayer. I prepare the new media on the created MediaPlayer and then set the EMPC as the new contentPane of the frame. But it does not work as I expected it to work yet. I am sure there is a way easier and better approach.
I am not sure how exactly to use the caching ability of libvlc but maybe that is the right direction?

I am really thankful for any kind of help.

**Comment by caprica** [comment]
If you use a MediaListPlayer and a play-list, I believe recent versions of VLC are indeed supposed to linger on the final frame of video as per --play-and-pause.

Generally speaking, your only chance of seamless playback when playing multiple sequential videos is if you use a MediaListPlayer and a play-list.

**Comment by caprica** [comment]
Another point, although unrelated, you should definitely not be creating a new media player component each time, you absolutely do want to be reusing those.

**Comment by caprica** [comment]
Here's a test for you to see if what you want to achieve is possible...

Open the VLC application, drop your videos into the playlist window, play the playlist, observe.

**Comment by Remonell** [comment]
I use MediaListPlayer now within a canvas to make the player go fullscreen on a specific screen(your testcode helped a lot). Sadly there are still black frames between videos. Also in the VLC Media Player itself it shows the same behaviour.
But I feel like there has to be a way that allowes me to achieve that when I program it myself. For example what if I place another canvas behind the mediaListPlayer. And when I want to play the next video it draws the frame of the video in that moment to the canvas behind the player and brings it to the front until the mediaReadyEvent is fired from the new video. I will try that out now.

Do you have any other ideas that could point me in the right direction?

**Comment by caprica** [comment]
Did you try the command-line option I mentioned, "--play-and-pause", VLC help says it has this effect: "Pause each item in the playlist on the last frame."

Maybe also look at options for buffering to see if you can disable it.

I have no idea if it will help you, but might be worth a quick test.

If play and pause worked, you could in principle capture the last frame of the video via the snapshot API and paint that as a BufferedImage on your other Canvas.

I dunno really.

**Comment by caprica** [comment]
I mean, right now it's not clear to me exactly what your black frames are - is it while buffering, is the media just slow to start.

Interestingly with VLC 4.0.0 (still in development) and vlcj-5 (still in development!), you will be able to get a thumbnail via the API (it can be whatever size you want) at 100% playback position (without playing the video) and you could paint that while you're waiting.

**Comment by Remonell** [comment]
Hey, sorry for the late response. Have been very buys with customer projects. I tried the "--play-and-pause" flag. But even without the flag the VLC Player stops at the last frame. With the flag it does the same and does not change the black frame behaviour. I did fiddle around with a few caching/buffering options. Didnt change anything sadly.

I will try to get the last frames from all the videos at the start of the software and paint them over another canvas as you suggested.

I think this thread can be closed. If I need more help, it will probably be worth to open a new issue for that. Thanks again for your help! Really apprecciate it!
